src/puppetlabs/puppetdb/honeysql.clj

```diff
@@ -0,0 +1,73 @@
+(ns puppetlabs.puppetdb.honeysql
+  "Some generic HoneySQL extensions, candidates for re-usability and
+  potential upstream submission."
+  (:require [honeysql.core :as hcore]
+            [honeysql.format :as hfmt]
+            [puppetlabs.puppetdb.schema :as pls]
+            [schema.core :as s])
+  (:import (honeysql.types SqlCall SqlRaw)))
+
+;; SCHEMA
+
+(def key-or-sql
+  "Either a honeysql call, or a keyword argument"
+  (s/either SqlCall SqlRaw s/Keyword))
+
+;; COMMMON DIRECT SQL FUNCTIONS
+
+(pls/defn-validated coalesce :- SqlCall
+  [& args :- [(s/one key-or-sql "Keyword") key-or-sql]]
+  "coalesce(arg, ...) sql function"
+  (apply hcore/call :coalesce args))
+
+(pls/defn-validated scast :- SqlCall
+  [source :- s/Keyword
+   target :- s/Keyword]
+  "cast(source AS target) sql function"
+  (hcore/call :cast (hcore/raw (str (hfmt/to-sql source)
```

Comment:
Looks like 0.5.2 has been released and includes my cast patch upstream https://github.com/jkk/honeysql/blob/0.5.2/src/honeysql/format.clj#L110